File: test/fixtures/scip-java-poc/src/main/java/com/example/services/UserService.java
package com.example.services;

import com.example.models.User;

public class UserService {
    public void register(User user) {
        if (validate(user)) {
            user.setName(user.getName().trim());
        }
    }

    public boolean validate(User user) {
        return !user.getName().trim().isEmpty();
    }
}

File: test/fixtures/scip-poc/Services/UserService.cs
using ScipPocSample.Models;

namespace ScipPocSample.Services;

public class UserService
{
    public void Register(User user)
    {
        if (Validate(user))
        {
            user.Name = user.Name.Trim();
        }
    }

    public bool Validate(User user)
    {
        return !string.IsNullOrWhiteSpace(user.Name);
    }
}
